lib/copy.go

```diff
@@ -37,13 +38,41 @@ func (a *ACBuild) Copy(from, to string) (err error) {
 
 	target := path.Join(a.CurrentACIPath, aci.RootfsDir, to)
 
-	dir, _ := path.Split(target)
-	if dir != "" {
-		err := os.MkdirAll(dir, 0755)
+	if explicitTarget {
+		if len(froms) != 1 {
+			return fmt.Errorf("can't copy multiple things to the same path")
+		}
+		dir, _ := path.Split(target)
+		if dir != "" {
+			err := os.MkdirAll(dir, 0755)
+			if err != nil {
+				return err
+			}
+		}
+
+		return fileutil.CopyTree(froms[0], target, uid.NewBlankUidRange())
```

Comment:
ok, the option help text suggests otherwise

On Mon, Nov 16, 2015 at 1:28 PM, Derek Gonyeo <notifications@github.com>
wrote:

> In lib/copy.go
> <https://github.com/appc/acbuild/pull/103#discussion_r44985907>:
>
> > -	dir, _ := path.Split(target)
> > -	if dir != "" {
> > -		err := os.MkdirAll(dir, 0755)
> > +	if explicitTarget {
> > +		if len(froms) != 1 {
> > +			return fmt.Errorf("can't copy multiple things to the same path")
> > +		}
> > +		dir, _ := path.Split(target)
> > +		if dir != "" {
> > +			err := os.MkdirAll(dir, 0755)
> > +			if err != nil {
> > +				return err
> > +			}
> > +		}
> > +
> > +		return fileutil.CopyTree(froms[0], target, uid.NewBlankUidRange())
>
> The source in this situation could still be a directory.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/appc/acbuild/pull/103/files#r44985907>.
>